models/issues/reaction.go

```diff
@@ -105,32 +168,29 @@ func findReactions(e db.Engine, opts FindReactionsOptions) ([]*Reaction, int64,
 	return reactions, count, err
 }
 
-func createReaction(e db.Engine, opts *ReactionOptions) (*Reaction, error) {
+func createReaction(ctx context.Context, opts *ReactionOptions) (*Reaction, error) {
 	reaction := &Reaction{
-		Type:    opts.Type,
-		UserID:  opts.Doer.ID,
-		IssueID: opts.Issue.ID,
+		Type:      opts.Type,
+		UserID:    opts.DoerID,
+		IssueID:   opts.IssueID,
+		CommentID: opts.CommentID,
 	}
 	findOpts := FindReactionsOptions{
-		IssueID:   opts.Issue.ID,
-		CommentID: -1, // reaction to issue only
+		IssueID:   opts.IssueID,
+		CommentID: opts.CommentID, // reaction to issue only
```

Comment:
```suggestion
		CommentID: opts.CommentID,
```

What does ` // reaction to issue only` really mean?